opm/simulators/timestepping/SimulatorTimer.cpp

```diff
@@ -114,7 +114,7 @@ namespace Opm
 
     boost::posix_time::ptime SimulatorTimer::currentDateTime() const
     {
-        return startDateTime() + boost::posix_time::seconds( (int) simulationTimeElapsed());
+        return startDateTime() + boost::posix_time::seconds( (double) simulationTimeElapsed());
```

Comment:
Actually, since simulationTimeElapsed() returns a double, you don't need the cast. Makes me wonder why it was used in the first place!

Reply:
probably to avoid some stupid compiler warning. if so, casting to`long long int` should help.

Reply:
If seconds() expects an integer type, we are limiting ourselves to 1-second timestep resolution, which is not ideal, perhaps we should use microseconds instead (if it's a 64 bit integer at least).

Reply:
When I removed the (double) the dates are messed up again.

Reply:
Sorry I don't think it was the removing of the (double) that messed it up. 
If I do: 

```boost::posix_time::minutes( simulationTimeElapsed() / 60); ```

it seems to work. Maybe it is time to make a test for this call?

Reply:
hm, looks like `boost::posix_time::{hours,minutes,seconds,...}` [expect `long` as their argument](http://www.boost.org/doc/libs/1_65_1/doc/html/date_time/posix_time.html#time_duration_count_constr). that's a bit unfortunate API design IMO, but we could lessen the impact of it by replacing above line with

```c++
 return startDateTime() + boost::posix_time::hours(static_cast<long>(simulationTimeElapsed()/3600)) + boost::posix_time::hours(static_cast<long>(std::fmod(simulationTimeElapsed(), 3600));
```

that's still not ideal because it extends the maximum time by "only" factor 3600. though.

Reply:
A `long` is 64 bits on all platforms we support, so it should have plenty resolution even at milliseconds (famous last words). We do not want to use (only) the minutes() helper, as that limits the shortest timestep to one minute (since it takes integer arguments). I think seconds() as in the current version of the PR is fine, but we could hit a problem with wanting to use shorter timesteps at some point.

@andlaus your suggestion is buggy, I assume you meant seconds() for the second call. In any case, I don't think it matters at all with 64 bits, do you?

Reply:
> A long is 64 bits on all platforms we support.

Right. I confused that with the fact that the normal integers are still 32 bits on x86 and that only `long long` is guaranteed to be 64 bits.

but in this case: 2^63 seconds is almost 300 billion years, so why does the current code go belly-up? (I assume @totto82 did not want to simulate until the end of the universe. ;))

> @andlaus your suggestion is buggy, I assume you meant seconds() for the second call. In any case, I don't think it matters at all with 64 bits, do you?

yes, that was a copy-and-pasto.

Reply:
> 2^63 seconds is almost 300 billion years, so why does the current code go belly-up?

I think boost internally has a 32 bit problem: https://github.com/OPM/opm-parser/pull/1092

Reply:
ok, to me this is a perfect reason to rip out boost::posix_time and replace it with [`std::chrono`](http://en.cppreference.com/w/cpp/chrono/duration). maybe I missed something, though.